sdk/scheduler/src/main/java/org/apache/mesos/offer/constrain/RegexMatcher.java

```diff
@@ -0,0 +1,57 @@
+package org.apache.mesos.offer.constrain;
+
+import java.util.regex.Pattern;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Implements fuzzy regular expression support for string comparisons.
+ */
+public class RegexMatcher implements StringMatcher {
+
+    public static StringMatcher create(String pattern) {
+        return new RegexMatcher(pattern);
+    }
+
+    private final Pattern pattern;
+
+    @JsonCreator
+    private RegexMatcher(@JsonProperty("pattern") String pattern) {
+        this.pattern = Pattern.compile(pattern);
+    }
+
+    @Override
+    public boolean match(String value) {
+        return pattern.matcher(value).matches();
```

Comment:
Looking at the `Pattern` interface, I kind of like `matches()` as the name for the current `match()` method, since it implies a return type of `boolean` rather than `void`. Might be worth changing if it's low-effort.